internal/pkg/midi/device/open_rgb.go

```diff
@@ -209,15 +286,63 @@ func findController(c *openrgb.Client, events map[string]bool) (openrgb.Device,
 			return openrgb.Device{}, 0, fmt.Errorf("resolve hidraw failed (%d/%d): %s", i, count, err)
 		}
 
-		if events[event] == true {
+		if events[event] {
 			return dev, i, nil
 		}
 	}
 
 	return openrgb.Device{}, 0, fmt.Errorf("controller not found")
 }
 
-func (d *Device) handleOpenrgb(wg *sync.WaitGroup, ctx context.Context) {
+// value range -1.0 - 1.0
+func shiftColor(color openrgb.Color, value float64) openrgb.Color {
+	c := colorful.Color{
+		R: float64(color.Red) / 255,
+		G: float64(color.Green) / 255,
+		B: float64(color.Blue) / 255,
+	}
+	h, s, v := c.Hsv()
+
+	hue := math.Mod(math.Mod(h+value*120, 360)+360, 360)
+
+	nc := colorful.Hsv(hue, s, v)
+	return openrgb.Color{
+		Red:   byte(nc.R * 255),
+		Green: byte(nc.G * 255),
+		Blue:  byte(nc.B * 255),
+	}
+}
+
+// value range -1.0 - 1.0
+func valueToColor(value, s, v float64) openrgb.Color {
+	value = 120 - value*120
+
+	c := colorful.Hsv(value, s, v)
+	return openrgb.Color{
+		Red:   byte(c.R * 255),
+		Green: byte(c.G * 255),
+		Blue:  byte(c.B * 255),
+	}
+}
+
+func (d *Device) handleOpenrgb(ctx context.Context, wg *sync.WaitGroup) {
+	// time.Sleep(time.Second) // todo remove
+	// still panics:
+	// panic: runtime error: index out of range [3] with length 0
+	//
+	// goroutine 73 [running]:
+	// encoding/binary.littleEndian.Uint32(...)
+	//        /home/pi/bin/go/src/encoding/binary/binary.go:64
+	// github.com/realbucksavage/openrgb-go.(*Client).GetControllerCount(0x4000288210?)
+	//        /home/pi/go/pkg/mod/github.com/realbucksavage/openrgb-go@v0.0.0-20220821164356-dc79903db082/client.go:52 +0x9c
+	// github.com/gethiox/HIDI/internal/pkg/midi/device.findController(0x0?, 0x0?)
+	//        /home/pi/HIDI/internal/pkg/midi/device/open_rgb.go:181 +0x48
+	// github.com/gethiox/HIDI/internal/pkg/midi/device.(*Device).handleOpenrgb(0x40002bac00, {0x890be8, 0x400007c300}, 0x40000242a0?)
+	//        /home/pi/HIDI/internal/pkg/midi/device/open_rgb.go:282 +0x910
+	// created by github.com/gethiox/HIDI/internal/pkg/midi/device.(*Device).ProcessEvents
+	//        /home/pi/HIDI/internal/pkg/midi/device/events.go:398 +0x24c
+	// panic: runtime error: index out of range [3] with length 0
```

Comment:
that stacktrace is very impressive, but what for?

Reply:
it relates to some relatively rare panic in 3rd party library (OpenRGB client). Removing for now, LGTM :+1: